src/psbt.cpp

```diff
@@ -249,7 +249,7 @@ void PSBTOutput::FromSignatureData(const SignatureData& sigdata)
     if (!sigdata.tr_spenddata.internal_key.IsNull()) {
         m_tap_internal_key = sigdata.tr_spenddata.internal_key;
     }
-    if (sigdata.tr_builder.has_value()) {
```

Comment:
let's add a test for this specific case

Reply:
I think this is not really testable since we would not have a `tr_builder` for change that doesn't have scripts. This could only have been reached through a bad psbt, which have now been disallowed through serialization, and this change is more of a belt-and-suspenders.

Reply:
seems this would only be hit internally, resulting in a psbt with less data. not too troubling